src/legacy/core_plugins/data/public/filter/filter_manager/filter_manager.ts

```diff
@@ -85,9 +85,17 @@ export class FilterManager {
   private handleStateUpdate(newFilters: Filter[]) {
     // global filters should always be first
     newFilters.sort((a: Filter, b: Filter): number => {
-      if (a.$state && a.$state.store === FilterStateStore.GLOBAL_STATE) {
+      // TypeScript, this shouldn't happen
+      if (!a.$state || !b.$state) return 0;
```

Comment:
I don't think we need this comment

```js
// TypeScript, this shouldn't happen
```

Checking if some optional property exists in TypeScript is common.

![image](https://user-images.githubusercontent.com/9773803/60719285-7b23de80-9f27-11e9-9d0b-7c9c897ad008.png)

If you don't want to write the `if` statement explicitly and you 100% know that property exists, you can use TypeScript's null assertion `!`, like so:

```ts
a.$state!.store
```